File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int best = 0;
        int min = prices[0];
        for(int i = 1; i < prices.length; i++){
            if(prices[i] < min) min = prices[i];
            else best = Math.max(best, prices[i] - min);
        }
        return best;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_profit = 0
        smallest = float('inf')
        for price in prices:
            if price < smallest:
                smallest = price
            else: 
                max_profit = max(max_profit, price - smallest)
        return max_profit